specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc

```diff
@@ -30,38 +30,35 @@ might nonetheless include support for the technology. In such a case,
 the requirements for that technology described in this chapter would
 apply.
 
+[[a3537]]
 ==== Java Compatible APIs
 
 The containers provide all application
 components with at least the Java Platform, Standard Edition, v8 (Java
 SE) APIs. Containers may provide newer versions of the Java SE platform,
-provided they meet all the Jakarta EE platform requirements. The Java SE
-platform includes the following enterprise technologies:
+provided they meet all the Jakarta EE platform requirements. If the container
+provides a version of the Java SE platform newer than Java SE 8 and that
+version no longer includes one or more APIs that were available in Java SE 8,
+the container is not required to provide those APIs, unless they are
+one of the enterprise technologies listed in this specification.
 
-* Java IDL footnote:javaremoval[Removed from Java SE 11. See <<a2161, Required Jakarta Technologies>>.]
+The Java SE platform includes the following enterprise technologies:
+
+* Java IDL footnote:javaidlremoval[Removed from Java SE 11. Support for Java IDL is optional (see <<a3539, Java IDL (Optional)>>.) Product vendors that support Java IDL on a Java SE version that does not provide the APIs associated with Java IDL must otherwise provide those APIs to application components.]
 * JDBC
 * RMI-JRMP
-* RMI-IIOP footnote:javaremoval[]
+* RMI-IIOP footnote:rmiiiopremoval[Removed from Java SE 11. Support for RMI-IIOP is optional (see <<a3538, RMI-IIOP (Optional)>>.) Product vendors that support RMI-IIOP on a Java SE version that does not provide the RMI-IIOP APIs must otherwise provide those APIs to application components. Product vendors that support the optional Enterprise Beans 2.x API group must ensure that the javax.rmi.PortableRemoteObject class is available to application components.]
```

Comment:
Similar comments as above for Java IDL...

`[Removed from Java SE 11. Support for RMI-IIOP is optional (see <<a3538, RMI-IIOP (Optional)>>.) Product vendors that wish to support RMI-IIOP on a Java SE version that does not provide the RMI-IIOP APIs must otherwise provide those APIs to application components. Product vendors that support the optional Enterprise Beans 2.x API group must ensure that the javax.rmi.PortableRemoteObject class is available to application components.]`

Reply:
Thanks; done.